src/vmspawn/vmspawn.c

```diff
@@ -311,6 +318,33 @@ static int parse_argv(int argc, char *argv[]) {
                                 return log_error_errno(r, "Failed to parse --secure-boot=%s: %m", optarg);
                         break;
 
+                case ARG_PRIVATE_USERS: {
+                        _cleanup_free_ char *buffer = NULL;
+                        const char *range, *shift;
+
+                        range = strchr(optarg, ':');
+                        if (range) {
+                                buffer = strndup(optarg, range - optarg);
+                                if (!buffer)
+                                        return log_oom();
+                                shift = buffer;
+
+                                range++;
+                                r = safe_atou32(range, &arg_uid_range);
+                                if (r < 0)
+                                        return log_error_errno(r, "Failed to parse UID range \"%s\": %m", range);
+                        } else
+                                shift = optarg;
+
+                        r = parse_uid(shift, &arg_uid_shift);
+                        if (r < 0)
+                                return log_error_errno(r, "Failed to parse UID \"%s\": %m", optarg);
```

Comment:
can you add a helper func for this and add it to basic/namespace-util.[ch], that parses such userns uid range definitions, and then share this with nspawn.